src/res/heatmap.html

```diff
@@ -0,0 +1,4289 @@
+        status.textContent = '\xa0';
+        status.style.display = 'none';
+        canvas.title = '';
+        canvas.style.cursor = '';
+        canvas.onclick = '';
+        if (e !== undefined) {
+            preview_wrapper.style.display = 'none';
+            previewX = previewY = -1;
+        }
+    }
+
+    document.getElementById('reset').onclick = function () {
+        search(false);
+    }
+
+    class DataBuffer {
+        data;
+        pos = 0;
+
+        constructor(encodedData) {
+            this.data = encodedData;
+        }
+
+        nextVarInt() {
+            let res = 0;
+            let shift = 1;
+            let b;
+            do {
+                b = this.byteAt(this.pos++);
+                res += (b % 61) * shift;
+                shift *= 61;
+            } while (b >= 61);
+            return res;
+        }
+
+        nextInt6() {
+            return this.byteAt(this.pos++);
+        }
+
+        nextSymbol() {
+            return this.byteAt(this.pos++);
+        }
+
+        nextInt30() {
+            return this.byteAt(this.pos++) << 0
+                | this.byteAt(this.pos++) << 6
+                | this.byteAt(this.pos++) << 12
+                | this.byteAt(this.pos++) << 18
+                | this.byteAt(this.pos++) << 24;
+        }
+
+        byteAt(pos) {
+            const c = this.data.charCodeAt(pos);
+            switch (c) {
+                case 127:
+                    return 0;
+                case 126:
+                    return 13;
+                case 125:
+                    return 38;
+                case 124:
+                    return 60;
+                case 123:
+                    return 62;
+            }
+            return c;
+        }
+
+        nextInt18() {
+            return this.nextInt6() | this.nextInt6() << 6 | this.nextInt6() << 12;
+        }
+
+        int30(pos) {
+            let p = pos * 5;
+            if (p < 0) {
+                p += this.data.length;
+            }
+            return this.byteAt(p++) << 0
+                | this.byteAt(p++) << 6
+                | this.byteAt(p++) << 12
+                | this.byteAt(p++) << 18
+                | this.byteAt(p++) << 24;
+        }
+
+    }
+
+    function decodeExecutions(data) {
+        const samplesCount = data.int30(-1);
+        const chunksCount = data.int30(-2);
+        const storageSize = data.int30(-3);
+        const blocksCount = data.int30(-4);
+        const lz78RecordsCount = data.int30(-5);
+
+        const starts = new Set();
+        const startMethodsCount = data.nextVarInt();
+        for (let i = 0; i < startMethodsCount; i++) {
+            starts.add(data.nextVarInt());
+        }
+
+        // NOTE: All dynamic allocations should be done before this line
+        const buffer = new ArrayBuffer((4 + 4 + 4 + 4 + 4) * lz78RecordsCount);
+        const lz78RootMethods = new Uint32Array(buffer, 0, lz78RecordsCount);   // FIXME NOTE!!! methods in most cases takes only 2 bytes!
```

Comment:
There are too many FIXMEs and TODOs in the code. Could you please cleanup it a bit?

Reply:
done, kind of)
most of those todo's are relevant, but i'm it is almost 0 chance to fix them someday...)